samples/boards/nrf52/mesh/onoff-app/src/main.c

```diff
@@ -0,0 +1,694 @@
+
+static struct gpio_callback button_cb;
+
+static u8_t trans_id;
+static u32_t time, last_time;
+static u16_t primary_addr;
+static u16_t primary_net_idx;
+
+/*
+ * Generic OnOff Model Server Message Handlers
+ *
+ * Mesh Model Specification 3.1.1
+ *
+ */
+
+static void gen_onoff_get(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	struct net_buf_simple *msg = NET_BUF_SIMPLE(2 + 1 + 4);
+	struct onoff_state *onoff_state = model->user_data;
+
+	printk("%s addr: 0x%04x onoff: 0x%02x\n"
+			, __func__, model->elem->addr, onoff_state->current);
+	bt_mesh_model_msg_init(msg, BT_MESH_MODEL_OP_GEN_ONOFF_STATUS);
+	net_buf_simple_add_u8(msg, onoff_state->current);
+
+	if (bt_mesh_model_send(model, ctx, msg, NULL, NULL)) {
+		printk("%s Unable to send On Off Status response\n",
+				__func__);
+	}
+}
+
+static void gen_onoff_set_unack(struct bt_mesh_model *model,
+				struct bt_mesh_msg_ctx *ctx,
+				struct net_buf_simple *buf)
+{
+	struct net_buf_simple *msg = model->pub->msg;
+	struct onoff_state *onoff_state = model->user_data;
+	int err;
+	u8_t local_trans_id;
+
+	/* Ignore command sent by us to us as the LED state is set by button */
+	if (model->elem->addr != ctx->addr) {
+		onoff_state->current = net_buf_simple_pull_u8(buf);
+		local_trans_id = net_buf_simple_pull_u8(buf);
+		printk("%s addr: 0x%02x state: 0x%02x trans_id:"
+				" 0x%02x src: 0x%04x\n",
+				__func__, model->elem->addr,
+				onoff_state->current,
+				local_trans_id, ctx->addr);
+	}
+
+	/* Pin set low turns on LED's on the nrf52840-pca10056 board */
+	gpio_pin_write(onoff_state->led_device,
+			onoff_state->led_gpio_pin,
+			onoff_state->current ? 0 : 1);
+
+	/* Only publish if there is an assigned address */
+	if (model->pub->addr == BT_MESH_ADDR_UNASSIGNED) {
+		return;
+	}
+
+	/*
+	 * If a server has a publish address, it is required to
+	 * publish status on a state change
+	 *
+	 * See Mesh Profile Specification 3.7.6.1.2
+	 */
+
+	if (onoff_state->previous != onoff_state->current) {
+		printk("%s publish last: 0x%02x cur: 0x%02x\n",
+				__func__, onoff_state->previous,
+				onoff_state->current);
+		onoff_state->previous = onoff_state->current;
+		bt_mesh_model_msg_init(msg,
+				BT_MESH_MODEL_OP_GEN_ONOFF_STATUS);
+		net_buf_simple_add_u8(msg, onoff_state->current);
+		err = bt_mesh_model_publish(model);
+		if (err) {
+			printk("%s bt_mesh_model_publish: err: %d\n",
+					__func__, err);
+		}
+	}
+}
+
+static void gen_onoff_set(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	printk("%s\n", __func__);
+
+	gen_onoff_set_unack(model, ctx, buf);
+	gen_onoff_get(model, ctx, buf);
+}
+
+static void gen_onoff_status(struct bt_mesh_model *model,
+			     struct bt_mesh_msg_ctx *ctx,
+			     struct net_buf_simple *buf)
+{
+	u8_t	state;
+
+	state = net_buf_simple_pull_u8(buf);
+
+	printk("Node 0x%04x received OnOff status from 0x%04x with state 0x%02x\n",
```

Comment:
Minor thing: you can eliminate the over 80-chars violation by dropping "received" here, since it's implied by "from"

Reply:
ok